### PR TITLE
Select remote and warning fixes

### PR DESCRIPTION
**Pull request by unmultimedio**
PR comments from https://github.com/bufbuild/buf/pull/2178

- Move Discover func out of `bufmoduleref` pkg, and rename to `SelectReferenceForRemote`.
- Remove Changelog entries.
- Improve debug message about unnamed module.

### CODE REVIEW

File: private/buf/bufcli/remote_discovery_test.go [hunk 1]
```diff
@@ -12,89 +12,95 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package bufmoduleref
+package bufcli
```
Comment (pkwarren) [inline private/buf/bufcli/remote_discovery_test.go:15]: Should this be rolled into the existing `bufcli_test.go`?

**Comment by unmultimedio** [comment]
Tested locally, where `buf` is the released CLI, and `bufl` is my local buf version.

```
$ buf mod update
WARN	"buf.yaml" does not specify a "name", so Buf is defaulting to using remote "buf.build" for dependency resolution. This remote may be unable to resolve "bufbuild.internal/jfigueroa/people" if it's an enterprise BSR module. Did you mean to specify a "name: bufbuild.internal/..." on this module?
Failure: invalid remote "bufbuild.internal" for dependency "bufbuild.internal/jfigueroa/people", did you mean "buf.build"?

$ bufl mod update

$ bufl mod update --debug
DEBUG	File "buf.yaml" does not specify the "name" field. Based on the dependency "bufbuild.internal/jfigueroa/people", it appears that you are using a BSR instance at "bufbuild.internal". Did you mean to specify "name: bufbuild.internal/..." within "buf.yaml"?
DEBUG	get_config	{"duration": "143µs"}
DEBUG	command	{"duration": "55.57625ms"}
```

**Review by pkwarren — COMMENTED** [review]
One comment - will leave for @bufdev approval.

File: private/buf/bufcli/remote_discovery_test.go (hunk 1 above)
Comment (unmultimedio) [reply private/buf/bufcli/remote_discovery_test.go]: Yup perfect, renamed the pkg, although in that file you mention we do openbox `bufcli` testing 😞

Comment (unmultimedio) [reply private/buf/bufcli/remote_discovery_test.go]: Also, moved the other tests to closed box, and moved everything to the same file.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,7 @@
 
 ## [Unreleased]
 
-- Remove an import warning in case the import statement points to a local unnamed module.
-- Discover the correct remote to use for fetching dependencies in the `mod update` and `mod prune`
-  commands, in case it's an unnamed module and there are a mix of remotes in the `buf.yaml`
-  configuration file.
```
Comment (unmultimedio) [inline CHANGELOG.md:8]: @bufdev I still think we should keep a changelog entry for this. `buf mod update` and `buf mod prune` behavior changed, and now we select the remote for dependencies more intuitively, allowing pro customers to keep local modules unnamed.

**Review by bufdev — APPROVED** [review]

**Pull request merged**